WcaOnRails/app/models/competition.rb

```diff
@@ -466,22 +466,76 @@ def country
   end
 
   def self.search(query, params: {})
-    sql_query = "%#{query}%"
-    Competition.where("id LIKE :sql_query OR name LIKE :sql_query OR cellName LIKE :sql_query OR cityName LIKE :sql_query OR countryId LIKE :sql_query", sql_query: sql_query).order(year: :desc, month: :desc, day: :desc)
+    competitions = Competition.where(showAtAll: true)
```

Comment:
> I get the attraction of an object that can both be a query and a result

What do you mean by that? A relation is immutable and calls to `to_a` _return_ the result and don't mutate the query/relation. So, there is no one object which can be both at different times. They don't even have the same interface, `User.all.to_a.where(id: 4)` doesn't work.

> though I hear there are better alternative Gems.

`sequel` is one of them, but essentially works the same way (build queries using method chaining/[fluent interface](https://en.wikipedia.org/wiki/Fluent_interface).

> If AR wasn't so primitive, you could use the much cleaner Hash syntax

Do you have a short code example? Maybe we could build something similar to get rid of the `if foo; relation = relation.where(...); end` boilerplate.